pkg/cluster/defaults.go

```diff
@@ -21,10 +26,55 @@ func NewControlPlaneIPCheckAnnotationDefaulter(skipIPCheck bool) ControlPlaneIPC
 }
 
 // ControlPlaneIPCheckDefault sets the annotation for control plane skip ip check if the flag is set to true.
-func (d ControlPlaneIPCheckAnnotationDefaulter) ControlPlaneIPCheckDefault(ctx context.Context, spec *Spec) (*Spec, error) {
+func (d ControlPlaneIPCheckAnnotationDefaulter) ControlPlaneIPCheckDefault(ctx context.Context, cluster *anywherev1.Cluster) (*anywherev1.Cluster, error) {
```

Comment:
So we want to change these to cluster now?

Reply:
yes, since we aren't yet using the cluster spec anywhere, changed it to cluster.

Reply:
we were going to build the spec from the controller, however since we don't use spec today in any of the defaulters, I decided to not build it unless an actual use case comes up where we want to default something other than the cluster object.

Reply:
since we are using just the cluster now and not updating any other objects in the spec and based on the discussion here https://github.com/aws/eks-anywhere/pull/6176#discussion_r1271315060 , this defaults also needs to be updated since all defaults need to have the same signature.

Reply:
I think this is the same as https://github.com/aws/eks-anywhere/pull/6176/files#r1273894766
I'll continue here and we can close the other one

IMO we should avoid having to build the spec in the controller if we don't need it. But we shouldn't change the CLI defaulter signature. That took a spec for a reason, it's explained in the validations and defaults design doc.